code/core/src/types/modules/core-common.ts

```diff
@@ -194,6 +194,7 @@ export interface BuilderOptions {
   ignorePreview?: boolean;
   cache?: FileSystemCache;
   configDir: string;
+  projectRoot?: string;
```

Comment:
can you elaborate on why this is needed now?

Reply:
It's not strictly necessary, but before the paths were relative to `cwd` (which might not be the project root). I've actually extracted the root directory actually only to make testing of the `toImportFn` simpler - but then thought that I would make sense to propagate this upwards and add a proper handling of the project root.
https://github.com/storybookjs/storybook/pull/28941/files#diff-a4ade24974a8c7f082261c73f22be07ec54ad61a3f9683b20035955a50fa1a8bR54

Reply:
I'm a little concerned that this doesn't actually solve the issue, but appears to. I wouldn't be surprised if there were hundreds of other places that assumed that `process.cwd() === projectRoot`, that would break if anyone attempted to use a project root different from the CWD. So then they'd set this explicitly, thinking that that would solve the issue, when it would just cause breakage somewhere else.

It's just my experience with the code base that makes me slightly defensive about this. We would need to set up an actual minimal reproduction with a case where CWD is not project root and see that it worked, before I'd feel confident in this.

I do like your thinking behind this though, on the surface it makes sense.

Reply:
Is this really a user-facing option? That was not my intention. I only set it once to be the parent of the config dir.
But sure, there might be other problems with not using the root as cwd.

Reply:
I'm unsure about it being user-facing, it's at least _readable_ by users, but I don't think it's writable.

Would still be great to confirm it works.